Documentation/ExtensionArchitecture/Extbase/Introduction/Index.rst

```diff
@@ -0,0 +1,92 @@
+.. include:: /Includes.rst.txt
+.. index::
+   pair: Extension development; Extbase introduction
+
+.. _extbase-introduction:
+
+====================
+Extbase introduction
+====================
+
+.. todo:: Refer to a good example, e.g. tea, blog_example?
+
+.. todo:: Add tips how to get more information or help
+
+What is Extbase?
+================
+
+Extbase is provided via a TYPO3 system extension (typo3-cms/extbase). Extbase is
+a framework within the TYPO3 Core. The framework is based on the software pattern
+`MVC (model-view-controller) <https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller>`__
+and uses
+`ORM (object relational modeling) <https://en.wikipedia.org/wiki/Object%E2%80%93relational_mapping>`__.
+
+Extbase can be and is often used in combination with the Fluid templating engine,
+but Fluid **can** also be used without Extbase. All things that can be done with
+Extbase, can also be done with TYPO3 Core native functionality.
+Extbase is not a prerequisite for extension development. In most cases,
+using Extbase means writing less code, but the performance may suffer.
+
+When Extbase was released, it was introduced as the modern way to program
+extensions and the "old" way (pibase) was propagated as outdated. When we look
+at this today, it is not entirely true: Extbase is a good fit for some specific
+types of extensions and there are always alternatives. For some use cases it
+is not a good fit at all and the extension can and should be developed without
+Extbase.
+
+Thus, many things, such as backend modules or plugins can be done "the Extbase
+way" or "the Core way". This is a design decision, the extension developer must
+make for the specific use case.
+
+.. important::
+
+   Extbase is one of many options in the "toolbox" of TYPO3 extension development.
+
+   TYPO3 is flexible, often provides several approaches for one thing and it is
+   up to the extension developer to make an informed choice.
+
+
+Extbase or not?
+===============
+
+When to use Extbase and when to use other methods?
+
+As a rule of thumb (which should not be blindly followed but gives some
+guidance):
+
+Use Extbase if you:
+
+*  wish to get started quickly, e.g. using the
+   :ref:`extension_builder <extension-builder>`
+*  are a beginner or do not have much experience with TYPO3
+*  want to create a "classic" Extbase extension with plugins and (possibly)
+   backend modules (as created with the extension_builder)
+
+Do not use Extbase
+
+*  if performance might be an issue with the "classic" Extbase approach
+*  if there is no benefit from using the Extbase approach
+*  if you are writing an extension where Extbase does not add any or much value,
+   e.g. an extension consisting only of Backend modules, a site package, a
+   collection of content elements, an Extension which is used as a command line
+   tool.
+
```

Comment:
Some other issues when using Extbase: If you want to create an extension that should be extended by other extensions you run into a typical ORM issue, where your domain model cannot be easily extended multiple times (single-table inheritance). Also if you want to hook into the request process early via PSR-15 middlewares, it is possible that Extbase Configuration functionality it cannot easily be loaded up during a PSR-15 middleware. In such a case, it might be a better fit to use a custom "page.typeNum = 12312312" logic for .e.g AJAX Request when Extbase is wanted.

Reply:
"extender" is a good third-party solution to the issue of extending domain models. https://extensions.typo3.org/extension/extender

Reply:
I find this a little too specific for the "Introduction". We could add some more examples but not sure if they should go in the introduction. I saved the comment for now as issue - to get back to later. https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/2135

Reply:
I think it would be good to put that in a tips and tricks section or a tutorial